api/proto/teleport/legacy/types/types.proto

```diff
@@ -6183,3 +6187,9 @@ message KubernetesMatcher {
     (gogoproto.customtype) = "Labels"
   ];
 }
+
+// OktaOptions specify options related to the Okta service.
+message OktaOptions {
+  // SyncPeriod is the duration between synchronization calls in nanoseconds.
+  int64 SyncPeriod = 1 [(gogoproto.jsontag) = "sync_period,omitempty"];
```

Comment:
Can we change SyncPeriod to accept duration string instead at least? Configuring this in nanoseconds is awkward, I don't think we do this anywhere else.

Reply:
We do this a lot in `types.proto`, actually:

```proto
message CommandLabelV2 {
  // Period is a time between command runs
  int64 Period = 1 [
    (gogoproto.jsontag) = "period",
    (gogoproto.casttype) = "Duration"
  ];
...
  // GracePeriod is a period during which old and new CA
  // are valid for checking purposes, but only new CA is issuing certificates.
  int64 GracePeriod = 6 [
    (gogoproto.jsontag) = "grace_period,omitempty",
    (gogoproto.casttype) = "Duration"
  ];
...
  // RetentionPeriod is the retention period for audit events.
  int64 RetentionPeriod = 14 [
    (gogoproto.jsontag) = "retention_period",
    (gogoproto.casttype) = "Duration",
    (gogoproto.nullable) = true
  ];
...
  // AWSIIDTTL is the TTL to use for AWS EC2 Instance Identity Documents used
  // to join the cluster with this token.
  int64 AWSIIDTTL = 3 [
    (gogoproto.jsontag) = "aws_iid_ttl,omitempty",
    (gogoproto.casttype) = "Duration"
  ];
...
  // ClientIdleTimeout sets global cluster default setting for client idle
  // timeouts.
  int64 ClientIdleTimeout = 1 [
    (gogoproto.jsontag) = "client_idle_timeout",
    (gogoproto.casttype) = "Duration"
  ];
```

I'm avoiding using the `gogoproto.CastType` here so as to avoid needing to excise it later, but IIRC `google/protobuf/duration.proto` doesn't work too well with our setup for whatever reason.

Reply:
I understand bit asking users to write

```yaml
okta:
  sync_period: 1800000000000
```

to indicate they want 30m sync interval is not great UX. I would prefer supporting duration like we do for other fields.

Reply:
Sounds good. I've made this change by adding in the `gogoproto.cast_type` in.